src/pudl_archiver/orchestrator.py

```diff
@@ -292,18 +248,24 @@ def _generate_changes(
             resource=resource.local_path,
         )
 
-    def _get_files_to_delete(self, resources) -> list[str]:
+    def _get_deletions(self, resources) -> list[_DepositionChange]:
         # Delete files not included in new deposition
         files_to_delete = []
-        for filename, file_info in self.deposition_files.items():
+        for filename in self.deposition_files:
             if filename not in resources and filename != "datapackage.json":
                 logger.info(f"Deleting {filename} from deposition.")
-                files_to_delete.append(filename)
+                files_to_delete.append(
+                    _DepositionChange(_DepositionAction.DELETE, name=filename)
+                )
 
         return files_to_delete
 
     async def _apply_change(self, change: _DepositionChange):
         """Actually upload and delete what we listed in self.uploads/deletes."""
+        self.changes.append(change)
+        if self.dry_run:
```

Comment:
Thought it would be good to move the dry-run checks closer to the actual write operations, but I didn't feel like making a whole bunch of changes to the depositor (or write a dry-run specific depositor). So here we are.